src/curve/arithmetic.rs

```diff
@@ -66,12 +66,12 @@ impl<P: SchemeParams> Scalar<P> {
         Self(backend_scalar)
     }
 
-    pub fn random(rng: &mut impl CryptoRngCore) -> Self {
-        Self(ScalarPrimitive::<P::Curve>::random(rng).into())
+    pub fn random(rng: &mut dyn CryptoRngCore) -> Self {
+        Self(ScalarPrimitive::<P::Curve>::random(&mut BoxedRng(rng)).into())
     }
 
-    pub fn random_nonzero(rng: &mut impl CryptoRngCore) -> Self {
-        Self(*NonZeroScalar::<P::Curve>::random(rng).as_ref())
+    pub fn random_nonzero(rng: &mut dyn CryptoRngCore) -> Self {
+        Self(*NonZeroScalar::<P::Curve>::random(&mut BoxedRng(rng)).as_ref())
```

Comment:
This is pretty annoying esthetically; it's pretty much just doing `Uint::random_mod(&mut rng, P::CURVE_ORDER)`. Sigh.